pkg/hintrunner/zero/zerohint_ec.go

```diff
@@ -1074,3 +1074,205 @@ func createRandomEcPointHinter(resolver hintReferenceResolver) (hinter.Hinter, e
+			//> from starkware.crypto.signature.signature import ALPHA, BETA, FIELD_PRIME
+			//> from starkware.python.math_utils import random_ec_point
+			//> from starkware.python.utils import to_bytes
+			//>
+			//> n_elms = ids.len
+			//> assert isinstance(n_elms, int) and n_elms >= 0, \
+			//> 	f'Invalid value for len. Got: {n_elms}.'
+			//> if '__chained_ec_op_max_len' in globals():
+			//> 	assert n_elms <= __chained_ec_op_max_len, \
+			//> 		f'chained_ec_op() can only be used with len<={__chained_ec_op_max_len}. ' \
+			//> 		f'Got: n_elms={n_elms}.'
+			//>
+			//> # Define a seed for random_ec_point that's dependent on all the input, so that:
+			//> #   (1) The added point s is deterministic.
+			//> #   (2) It's hard to choose inputs for which the builtin will fail.
+			//> seed = b"".join(
+			//> 	map(
+			//> 		to_bytes,
+			//> 		[
+			//> 			ids.p.x,
+			//> 			ids.p.y,
+			//> 			*memory.get_range(ids.m, n_elms),
+			//> 			*memory.get_range(ids.q.address_, 2 * n_elms),
+			//> 		],
+			//> 	)
+			//> )
+			//> ids.s.x, ids.s.y = random_ec_point(FIELD_PRIME, ALPHA, BETA, seed)
+
+			nElms, err := hinter.ResolveAsUint64(vm, len)
+			if err != nil {
+				return err
+			}
+
+			if nElms == 0 {
+				return fmt.Errorf("invalid value for len. Got: %v", nElms)
+			}
+
+			chainedEcOpMaxLen := uint64(1000)
+			if nElms > chainedEcOpMaxLen {
+				return fmt.Errorf("f'chained_ec_op() can only be used with len<=%d.\n Got: n_elms=%d", chainedEcOpMaxLen, nElms)
+			}
+
+			pAddr, err := p.GetAddress(vm)
+			if err != nil {
+				return err
+			}
+
+			pValues, err := vm.Memory.ResolveAsEcPoint(pAddr)
+			if err != nil {
+				return err
+			}
+
+			mAddr, err := hinter.ResolveAsAddress(vm, m)
+			if err != nil {
+				return err
+			}
+
+			qAddr, err := hinter.ResolveAsAddress(vm, q)
+			if err != nil {
+				return err
+			}
+
+			firstRange, err := vm.Memory.GetConsecutiveMemoryValues(*mAddr, int16(nElms))
+			if err != nil {
+				return err
+			}
+
+			var firstRangeFelts []*fp.Element
+			for _, element := range firstRange {
+				firstRangeFelts = append(firstRangeFelts, &(element.Felt))
+			}
+
+			secondRange, err := vm.Memory.GetConsecutiveMemoryValues(*qAddr, int16(2*nElms))
+			if err != nil {
+				return err
+			}
+
+			var secondRangeFelts []*fp.Element
+			for _, element := range secondRange {
+				feltCopy := element.Felt
+				secondRangeFelts = append(secondRangeFelts, &feltCopy)
+			}
+
+			var bytesArray []byte
+
+			writeFeltToBytesArray := func(n *fp.Element) {
+				for _, byteValue := range n.Bytes() {
+					bytesArray = append(bytesArray, byteValue)
+				}
+			}
+
+			for _, felt := range pValues {
+				writeFeltToBytesArray(felt)
+			}
+			for _, felt := range firstRangeFelts {
+				writeFeltToBytesArray(felt)
+			}
+			for _, felt := range secondRangeFelts {
+				writeFeltToBytesArray(felt)
+			}
+
+			fmt.Println(bytesArray)
+
+			seed := sha256.Sum256(bytesArray)
```

Comment:
From here on, isn't this the same logic as in randomEcPoint? Could we have it in a separate method?

Reply:
yep it is, we can do that!

Reply:
done !